src/integrationtest/java/com.microsoft.aad.msal4j/AcquireTokenSilentIT.java

```diff
@@ -218,6 +219,52 @@ public void acquireTokenSilent_ConfidentialClient_acquireTokenSilentDifferentSco
                 .get();
     }
 
+    @Test(dataProvider = "environments", dataProviderClass = EnvironmentsProvider.class)
```

Comment:
I think it would be better to have use unit test to test this scenario, current test does not cover end to end scenario , also it is hybrid of unit test + integration testing , i would keep them separated

Reply:
Without a way to get `refresh_in` in an actual token response, I'm not sure if there's a way to have meaningful tests without a hybrid like this. A unit test of setting the `refreshOn` field would just be 'set `refreshOn` manually, then assert that it was set'. The only time it's used is halfway through the acquireTokenSilent flow, and a unit test of that would need a lot of hardcoding that the actual token request process takes care of anyway.